docs/enhancements/cluster_pool_hot_spares.md

```diff
@@ -0,0 +1,171 @@
+
+Add a field to ClusterPool indicating a portion of the pool to maintain in Running state.
+`spec.runningCount` is an integer absolute number of Running clusters to strive for.
+It is constrained to `0 <= runningCount <= Size`.
+We could enforce this by webhook; or we could just update the spec accordingly (explicitly set `runningCount = min(runningCount, size)`).
+
+If unspecified, the default behavior is to hibernate all CDs in the pool, as today.
+This corresponds to `runningCount=0`, conveniently the golang nil value of an int.
+
+### Decouple PowerState on Creation
+
+Today we create all new ClusterDeployments with `powerState=Hibernating` and update that to `Running` at claim time.
+With this feature, we will create with `powerState` unset.
+Then in a separate code path:
+- Sort unassigned (installing + ready) CDs by `creationTimestamp`
+- Update the oldest `runningCount` CDs with `powerState=Running`
+- Update the remainder (the newest `size - runningCount`) with `powerState=Hibernating`
+
+To be safe, and to continue correctly supporting pools with no/zero `runningCount`, we will still always set `powerState=Running` when assigning a claim.
+
+(In both cases, we should only perform an actual `Update()` for a given CD if its state is changed.)
+
+(Note: This means it is possible for a CD's PowerState to be updated several times even while it's installing.
+That's fine -- though we should make sure the hibernation controller handles it sanely.)
+
+### ClusterClaim Assignment by `creationTimestamp`
+
+When determining which ClusterDeployments to assign to new claims, we can simply sort the list by `creationTimestamp`.
+Since we're [prioritizing PowerState the same way](#decouple-powerstate-on-creation), this should serve to satisfy both:
+- FIFO behavior. Oldest clusters will be claimed first.
+- Minimum response time.
+  We always activate the oldest cluster, and we always claim the oldest cluster, so the first claim will pick up the least-recently-resumed cluster, which is the most likely to be Running, or will be there the soonest.
+
+Caveats:
+- Due to variability in install or resume time, it is technically possible that there would be a "better" (faster) choice for a given assignment.
+  But it'll be real close; and we prefer to code more strictly to the FIFO behavior.
+- Manually resuming a pool CD will not cause it to be claimed earlier.
+  In fact, it is subject to being re-hibernated by the controller.
+  This is an acceptable design point: You can manage power states by configuring your ClusterPool; don't expect to be able to do so manually.
+
+### Hibernation Controller Must Account For `hibernateAfter` in Pool Clusters
+
+If the ClusterPool has `hibernateAfter` set, that clock starts on a given ClusterDeployment from the time its
+[Hibernating condition is Running](https://github.com/openshift/hive/blob/28f67bab3dac5fb9dec405e0536d64187faf964d/pkg/controller/hibernation/hibernation_controller.go#L240-L244);
```

Comment:
Will soon be a Running condition = true for this.

Reply:
This all felt wrong as I started prototyping. This section proposes a new Condition on ClusterDeployment just so we can tell the claim time.

As noted here, the times the controller looks at today are last resume and when installation completed.
Last resume is in a Condition.
Installation completion is directly in a status field.
Resuming is a thing that can happen over and over.
Installation completes once; that field should never get updated again.

Claiming, like install completion, is a thing that happens once. So it makes more sense for it to be a status field.

So I'd like to change this to propose `status.claimedTimestamp` instead, and use it if set.

(It also occurs to me that I should describe what happens on "upgrade": We'll never set status.claimedTimestamp for CDs that have flown the coop, but that's okay because they also wouldn't have been subject to `runningCount`, so would have been resumed from hibernating when claimed, so would be using the last resumed time as the baseline for `hibernateAfter` anyway.)